src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -39,23 +51,55 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
         $this->joinRelations($queryBuilder, $resourceClass);
     }
 
+    public function getMetadataProperties($resourceClass): array {
+        $properties = [];
+
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass) as $property) {
+            $properties[$property] = $this->propertyMetadataFactory->create($resourceClass, $property);
+        }
+
+        return $properties;
+    }
+
     /**
      * Left joins relations to eager load.
      *
      * @param QueryBuilder $queryBuilder
      * @param string       $resourceClass
      */
-    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass)
+    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, string $originAlias = 'o', string &$relationAlias = 'a')
```

Comment:
It doesn't.

EagerLoadingExtension is effective before FilterExtension. If a join has been done there already, it was doing the join twice. Joining once is done in [`AbstractFilter`](https://github.com/api-platform/core/pull/717/commits/b9e04b451328bd2a4c1e4285c518dc43432d2975#diff-756b3787605765d5c843f78c389e03feR244).